enterprise/api_client/discovery.py

```diff
@@ -121,6 +121,29 @@ def get_search_results(self, querystring=None, traverse_pagination=True):
             many=False,
         )
 
+    def get_catalog_results(self, query):
```

Comment:
Instead of using a different method name how about something like `get_search_results_with_post_method` (like method above `get_search_results`)?

Reply:
Getting data using HTTP POST is the internal detail, I would prefer to name the method as it is.